cabal-install/Distribution/Client/ProjectOrchestration.hs

```diff
@@ -223,8 +223,13 @@ data ProjectBuildContext = ProjectBuildContext {
 runProjectPreBuildPhase
     :: Verbosity
     -> ProjectBaseContext
-    -> (ElaboratedInstallPlan -> IO ElaboratedInstallPlan)
-    -> IO ProjectBuildContext
+    -- ^ A function that selects a subset of the plan and returns it,
+    -- It can also return some arbitrary data, which can be useful in case
+    -- the caller needs it.
+    -> (ElaboratedInstallPlan -> IO (ElaboratedInstallPlan, a))
```

Comment:
Well, if I add another field it has to be `Maybe (UnitId, ComponentId)` or equivalent, because that record is used in other commands too. But then I'd have to unwrap that `Maybe` and rely on the assumption that I set it to  `Just` in preBuild.